### ISSUE TITLE
keras.backend.learning_phase() broken when sample_weights are used

### ISSUE THREAD

**Issue by pavanky** [tensorflow/tensorflow#32409]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
Yes

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Ubuntu 18.04

- TensorFlow installed from (source or binary):
binary

- TensorFlow version (use command below):
1.14.0

- Python version:
3.7

**Describe the current behavior**

There are multiple issues:

- *Issue 1* keras.backend.learning_phase() does not always return a scalar value (as stated by the [documentation](https://www.tensorflow.org/api_docs/python/tf/keras/backend/learning_phase) but it returns multiple values.
- *Issue 2* Even when though it returns multiple values, the shape says it is a 0d tensor indicating it is a scalar
- *Issue 3* Reduce does not do anything even though the tensor has multiple values.

**Describe the expected behavior**

- *Issue 1* Documentation is fixed to reflect the correct behavior
- *Issue 2* The right shape is set for the tensor
- *Issue 3* `reduce_any` should reduce all values when no axis is given irrespective of the shape.

**Code to reproduce the issue**
```
import os
import numpy as np
import tensorflow as tf
import tensorflow.keras as keras

BATCH_SIZE = 4

class SimpleModel(keras.Model):

  def __init__(self):
    super(SimpleModel, self).__init__()
    self._layer = keras.layers.Dense(1)

  def call(self, inputs):
    learning_phase = keras.backend.learning_phase()
    inputs = tf.Print(inputs, [tf.constant(learning_phase.shape.ndims)],
                      "Looking at the ndims says this is a salar: ",
                      summarize=BATCH_SIZE)
    inputs = tf.Print(inputs, [learning_phase], "But this is not a scalar: ",
                      summarize=BATCH_SIZE)
    inputs = tf.Print(inputs,
                      [tf.reduce_any(learning_phase)],
                      "Even reduce_any does not create a scalar: ", summarize=BATCH_SIZE)
    return self._layer(inputs)


def get_dataset(batch_size=BATCH_SIZE):
  num_batches = 16
  num_features = 5
  inputs = np.random.random((num_batches, num_features))
  labels = np.random.random((num_batches))
  sample_weights = (np.random.random((num_batches)) > 0.5).astype(np.float32)
  dataset = tf.data.Dataset.from_tensor_slices((inputs, labels, sample_weights))
  return dataset.batch(batch_size).take(1)


if __name__ == "__main__":
  tf.logging.set_verbosity(tf.logging.ERROR)
  os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'

  print("-------------")
  print(tf.__version__)
  print("-------------")

  model = SimpleModel()
  optimizer = tf.keras.optimizers.SGD(learning_rate=0.005)
  loss = tf.keras.losses.BinaryCrossentropy(from_logits=True)
  model.compile(optimizer, loss)
  model.fit(get_dataset(), verbose=0)
```

**Other info / logs**
```
-------------
1.14.0
-------------
Looking at the ndims says this is a salar: [0]
But this is not a scalar: [0 1 1 0]
Even reduce_any does not create a scalar: [0 1 1 0]
```

**Comment by jvishnuvardhan** [comment]
@pavanky Issue2 and 3 depends on Issue1 right? If issue1 is no longer persists, then Issue2 and 3 are no longer valid i guess. Please let us know what you think. You could close the issue and open a new issue when you can reproduce issue 2 and 3. Thanks!
